release/create_candidate_pr.py

```diff
@@ -0,0 +1,61 @@
+# pylint: disable=redefined-outer-name, invalid-name
+
+"""
+Creates a new release candidate branch and makes a pull request
+for it into the 'release' branch.
+"""
+
+from __future__ import print_function
+
+import argparse
+
+from release.github_api import GithubApi
+from release import utils
+
+
+def _build_parser():
+    """ Creates an option parser for this command """
+    expected_date = utils.default_expected_release_date()
+
+    result = argparse.ArgumentParser(
+        description="""
+        Create a new release candidate branch and associated pull request.
+        """
+    )
+
+    result.add_argument(
+        '--release-date', nargs=1, default=expected_date,
+        help="""
+        Specify a date that the release branch is expected to be deployed.
+        Should be in YYYY-MM-DD format. If not passed, defaults to the
+        next upcoming Tuesday, which is currently {date}.
+        """.format(date=expected_date.date().isoformat()),
+        )
+    result.add_argument(
+        '--org', nargs=1, default='edx',
+        help="""
+        Specify a github organization to work under. Default is 'edx'.
+        """)
+    result.add_argument(
+        '--repo', nargs=1, default='edx-platform',
+        help="""
+        Specify a github repository to work with. Default is 'edx-platform'.
+        """)
+
+    return result
+
+parser = _build_parser()
+args = parser.parse_args()
+
+github_api = GithubApi(args.org, args.repo)
+
+print("Fetching commits...")
+commit_hash = utils.most_recent_good_commit(github_api)
+branch_name = "rc/{date}".format(date=args.release_date.date().isoformat())
+
+print("Branching {rc} off {sha}".format(rc=branch_name, sha=commit_hash))
+github_api.create_branch(branch_name, commit_hash)
+
+print("Creating Pull Request for {rc} into master".format(rc=branch_name))
+request_title = "Release Candidate {rc}".format(rc=branch_name)
+github_api.create_pull_request(branch_name, title=request_title)
```

Comment:
Well, you can do a try/except that makes it easier to understand:
```python
try:
    github_api.create_pull_request(branch_name, title=request_title)
except Exception as exc:
    print("Some helpful message to the user explaining in plaintext the step that failed")
    raise exc
```

This allows you to explain more gracefully exactly what went wrong, and then re-raise the caught error.